--- origsrc/phonon-vlc/CMakeLists.txt	2009-07-16 11:39:56.000000000 -0500
+++ src/phonon-vlc/CMakeLists.txt	2009-07-16 12:39:14.363095200 -0500
@@ -25,11 +28,11 @@ set(phonon_vlc_SRCS
 )
 
 automoc4(phonon_vlc phonon_vlc_SRCS)
-add_library(phonon_vlc SHARED ${phonon_vlc_SRCS})
+add_library(phonon_vlc MODULE ${phonon_vlc_SRCS})
 
 set_target_properties(phonon_vlc PROPERTIES DEFINE_SYMBOL MAKE_PHONON_VLC_LIB)
 set_target_properties(phonon_vlc PROPERTIES PREFIX "")
-target_link_libraries(phonon_vlc ${PHONON_LIBS} ${VLC_LIBRARIES})
+target_link_libraries(phonon_vlc ${PHONON_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTCORE_LIBRARY} ${VLC_LIBRARIES})
 
 install(TARGETS phonon_vlc DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/phonon_backend)
 install(FILES vlc.desktop DESTINATION ${SERVICES_INSTALL_DIR}/phononbackends)
