--- origsrc/phonon-vlc/CMakeLists.txt	2009-07-16 11:39:56.000000000 -0500
+++ src/phonon-vlc/CMakeLists.txt	2009-07-16 12:39:14.363095200 -0500
@@ -1,8 +1,11 @@
 project(phonon-vlc)
 
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}")
+find_package(KDE4 REQUIRED)
+find_package(Phonon REQUIRED)
 find_package(VLC REQUIRED)
 
-include_directories(${QT_INCLUDES} ${VLC_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(${QT_INCLUDES} ${PHONON_INCLUDE_DIR} ${VLC_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR})
 
 add_definitions(-D_PHONON_BACKEND_VERSION_4_2)
 
